python/cudf/cudf/tests/test_string.py

```diff
@@ -1512,6 +1535,21 @@ def test_strings_rsplit(data, n, expand):
     )
 
 
+@pytest.mark.parametrize(
+    "data", [["a b", " c ", "   d", "e   ", "f"]],
+)
+@pytest.mark.parametrize("n", [-1, 0, 1, 3, 10])
+@pytest.mark.parametrize("expand", [True, False, None])
+def test_string_rsplit_re(data, n, expand):
+    ps = pd.Series(data, dtype="str")
+    gs = cudf.Series(data, dtype="str")
+
+    # Pandas does not support the regex parameter until 1.4.0
+    expect = ps.str.rsplit(pat=" ", n=n, expand=expand)
```

Comment:
Yea, that sounds like a better approach by using `PANDAS_LT_140` from [here](https://github.com/rapidsai/cudf/blob/branch-22.04/python/cudf/cudf/core/_compat.py#L12)

Reply:
This check is actually incorrect right now. While the pandas API documentation for `Series.str.split` includes the regex parameter, as @davidwendt pointed out the function signature doesn't actually support it. However, I just discussed with @shwina offline and we agree that this seems to be an oversight by pandas, especially since `split` does support the regex and the docstring indicates that they _intended_ to support it. We decided that the best path forward is
1. Open a pandas issue on this discrepancy between the API documentation and the actual implementation of `rsplit`.
2. Rewrite this test to manually verify the output rather than comparing to pandas for the cases that pandas does not currently support.

If we want to be fancy we could also make those test cases automatically fail if we detect that the parameters have been added by pandas. In other words, add something like this to the test:
```
# Once this assertion fails, update this test to validate against pandas for sufficiently new pandas versions.
assert 'regex' not in inspect.signature(pd.Series.str.rsplit).parameters.keys()
```

Reply:
CC @davidwendt

Reply:
For (1): The `rsplit` with regex issue appears to already be an open Pandas issue here: https://github.com/pandas-dev/pandas/issues/29633

Reply:
For (2): Would it be ok to compare Pandas `rsplit(' ')` to cudf `rsplit('\\s', regex=True)` since these should give equivalent results? 
```
    expect = ps.str.rsplit(pat=" ", n=n, expand=expand)
    got = gs.str.rsplit(pat="\\s", n=n, expand=expand, regex=True)
    assert_eq(expect, got)
```
Or are you looking for something more manual -- checking combinations of `n` and `expand` against a fixed set of Pandas Series or DataFrames?

Reply:
Got it. I think we're fine with (1) then. Regarding (2), if you are testing a particular split that is easily replicated without a regex that sounds fine to me. I thought you might be testing regex splits that can't be replicated with pure string splits, but if not then this seems fine. I don't think you particularly need to test complex regexes since we have plenty of verification on other code paths that the regex code works.

Reply:
Ok, thanks. I committed the above change to the pytest. Should I wait for another approval or can this be merged?